src/Integrations/Buckaroo/Buckaroo.php

```diff
@@ -0,0 +1,620 @@
+	 * @param  string $purchaseid      An ID that identifies the emandate with a purchase order. This will be shown in the emandate information of the customers' bank account. Max. 35 characters.
+	 * @param  string $language        The consumer language code in lowercase letters. For example `nl`, not `NL` or `nl-NL`.
+	 * @param  string $issuer          Issuer (bank) name.
+	 * @param  string $emandatereason  Description of the emandate.
+	 * @return array
+	 *
+	 * @throws BuckarooRequestException When something is wrong with response we get from Buckaroo.
+	 */
+	public function createEmandate(string $debtorreference, string $sequencetype, string $purchaseid, string $language, string $issuer, string $emandatereason): array
+	{
+		$response             = [];
+		$postArray           = $this->buildPostBodyForEmandate($debtorreference, $sequencetype, $purchaseid, $language, $issuer, $emandatereason);
+		$authorizationHeader = $this->generateAuthorizationHeader($postArray, $this->getBuckarooUri());
+
+		$postResponse = $this->guzzleClient->post("https://{$this->getBuckarooUri()}", [
+			'headers' => [
+				'Content-Type' => 'application/json',
+				'Authorization' => $authorizationHeader,
+			],
+			'body' => \wp_json_encode($postArray),
+		]);
+
+		$postResponseJson = json_decode((string) $postResponse->getBody(), true);
+
+		if (json_last_error() !== JSON_ERROR_NONE) {
+			throw new BuckarooRequestException(esc_html__('Invalid JSON in response body', 'eightshift-forms'));
+		}
+
+		if (! isset($postResponseJson['RequiredAction']['RedirectURL'])) {
+			throw new BuckarooRequestException(esc_html__('Missing redirect URL in Buckaroo response', 'eightshift-forms'), $postResponseJson);
+		}
+
+		$response['redirectUrl'] = $postResponseJson['RequiredAction']['RedirectURL'];
+
+		return $response;
+	}
+
+	/**
+	 * Creates a payment request.
+	 *
+	 * @param  int|float|string $donationAmount Donation amount.
+	 * @param  string           $invoice         Invoice name.
+	 * @param  string           $issuer          Issuer (bank) name.
+	 * @param  bool             $isRecurring    Is recurring payment.
+	 * @param  string           $description     Description of the payment.
+	 * @return array
+	 *
+	 * @throws BuckarooRequestException When something is wrong with JSON we get from Buckaroo.
+	 */
+	public function sendPayment($donationAmount, string $invoice, string $issuer, bool $isRecurring, string $description): array
+	{
+		$response             = [];
+		$postArray           = $this->buildPostBodyForPayment($donationAmount, $invoice, $issuer, $isRecurring, $description);
+		$authorizationHeader = $this->generateAuthorizationHeader($postArray, $this->getBuckarooUri());
+
+		$postResponse = $this->guzzleClient->post("https://{$this->getBuckarooUri()}", [
+		'headers' => [
+		'Content-Type' => 'application/json',
+		'Authorization' => $authorizationHeader,
+		],
+		'body' => \wp_json_encode($postArray),
+		]);
```

Comment:
Something odd happened here with the formatting 👀

Reply:
This happened everywhere for some reason because of PHPCBF (when it switched from spaces to tabs). Had to fill all instances of that manually. 😅